backend/common/utils/dataclass.py

```diff
@@ -16,5 +16,7 @@ def convert_dataclass_to_dict(data):
         for key, value in data.items():
             if is_dataclass(value):
                 data[key] = asdict(value)
+            elif isinstance(value, list):
```

Comment:
adding case where the dictionary to be converted contains a list of dataclasses as values